### ISSUE TITLE
cpplint gives bogus error on rvalue reference

### ISSUE THREAD

**Issue by janusw** [google/styleguide#61]
On the following C++ code

``` C++
#include <string>

int main() {
  struct A {
    A(std::string s, A&& a);
  };
}
```

cpplint complains:

```
rvalue_ref.cpp:5:  Missing spaces around &&  [whitespace/operators] [3]
```

It seems that the rvalue reference is misinterpreted as an operator. When making s an int, this turns into the expected:

```
rvalue_ref.cpp:5:  RValue references are an unapproved C++ feature.  [build/c++11] [3]
```

**Comment by tkruse** [comment]
I believe I have a valid patch in my fork of cpplint at https://github.com/tkruse/cpplint/commit/f23a35db3d693

**Comment by janusw** [comment]
@tkruse: Just tried your fork, and indeed it fixes the original test case. However, it still fails on variants like this:

``` c++
#include <string>

int main() {
  struct A {
    A(const std::string &s, A&& a);
  };
}
```

Thanks for working on this!

**Comment by tkruse** [comment]
thanks for the quick response. I included those situations as well now:
https://github.com/tkruse/cpplint/tree/fix-rvalueref-whitespace

**Comment by janusw** [comment]
Great, thanks. That fixes almost all instances of this problem that I see on my code. There is only a single case left:

``` c++
struct A {};
void create_output(const char *c,
                   A&& a);
```

**Comment by tkruse** [comment]
Hm. I could try to fix that, but honestly that would be much more hacking than the previous fixes, and the cpplint code is horrible enough as it is. (And my time is limited.) Is it not viable for you to refactor those into a single line or to add `// NOLINT`?

**Comment by janusw** [comment]
If this is very hard to fix, I can certainly work around it for now. Your other fixes are already a huge improvement.

(I have no clue about the internals of cpplint, but I'd expect that it has some sort of general mechanism to cope with line continuations. This kind of problem certainly arises in other situations as well...?)

**Comment by tkruse** [comment]
> but I'd expect that it has some sort of general mechanism to cope with line continuations. 

Sadly, it hasn't. cpplint is not a clean implementation, it mostly works line by line using regular expressions to guess what operator symbols might mean.

In some places the code goes back a few lines iteratively to improve the guessing, but not in a generic way.

**Comment by tkruse** [comment]
I believe this issue can be closed now after 01e47236c846571

**Comment by vapier** [comment]
public cpplint development in this repository has been halted (see #837).  please migrate to https://github.com/cpplint/cpplint now.
